### PR TITLE
channels: Add track support to commands.

### PR DESCRIPTION
**Pull request by josepht**
- status
  - close
  - revisions

LP: #1667318

### CODE REVIEW

**Comment by josepht** [comment]
This is not ready to merge until 'revisions' is finished.

**Comment by josepht** [comment]
depends on https://github.com/snapcore/snapcraft/pull/1160

File: snapcraft/storeapi/__init__.py [hunk 1]
```diff
@@ -635,7 +635,7 @@ def snap_revisions(self, snap_id, series, arch):
             qs['series'] = series
         if arch:
             qs['arch'] = arch
-        url = 'snaps/' + snap_id + '/history'
+        url = 'snaps/' + snap_id + '/revisions'
```
Comment (sergiusens) [inline snapcraft/storeapi/__init__.py]: why is this changed here?

Comment (josepht) [inline snapcraft/storeapi/__init__.py]: I got too eager.  Fixed.  This should explain why `revisions` and `list-revisions` was not working for you.

**Comment by sergiusens** [comment]
did you test this againts production? A manual test at least once?

**Comment by josepht** [comment]
Yes.

```
josepht@maple:~/src/git/snapcraft$ ./bin/snapcraft list-revisions hello-joe
Rev.    Uploaded              Arch          Version    Channels
5       2017-02-07T18:06:25Z  armhf         0.4        edge*, 0.1/edge*
4       2017-02-07T17:31:44Z  amd64, armhf  0.3        edge*, edge*
3       2017-02-07T16:18:00Z  amd64         0.3        0.2/edge*, 0.2/beta*, 0.2/edge*, 0.2/edge*, 0.2/edge*
2       2017-02-06T14:25:49Z  amd64         0.2        0.1/stable*, 0.1/edge*, 0.1/beta*, 0.1/edge*, 0.1/beta*, beta*
1       2017-02-03T15:16:54Z  amd64         0.1        0.1/edge, 0.1/edge, 0.1/edge, 0.1/edge, 0.1/edge, 0.1/edge, 0.1/edge, 0.1/edge, 0.1/edge, 0.1/edge, 0.1/edge, 0.1/edge, 0.1/edge, 0.1/edge, 0.1/edge, 0.1/beta, 0.2/edge, 0.2/edge, 0.2/edge, 0.2/edge
```
